src/key.rs

```diff
@@ -1057,9 +1057,13 @@ impl str::FromStr for KeyPair {
     type Err = Error;
 
     fn from_str(s: &str) -> Result<Self, Self::Err> {
-        let ctx = unsafe {
-            Secp256k1::from_raw_all(ffi::secp256k1_context_no_precomp as *mut ffi::Context)
-        };
+        #[cfg(feature = "global-context")]
+        let ctx = SECP256K1;
+
+        #[cfg(not(feature = "global-context"))]
+        let ctx = Secp256k1::signing_only();
```

Comment:
I wish I could deduplicate that code, but the global context is of type `Secp256k1<All>` while we apparently only need a signing context, so I can't easily build a function returning `Cow<'static, Secp256k1<_>>` (it would change its signature depending on the feature flags which is bad imo).

How much slower is creating an `All` context?

Reply:
I don't understand the question. What code needs to be deduplicated?

Reply:
The code highlighted with the comment which initializes `cfg` depending on the feature set. At first I had a `get_context() -> Cow<'static, Secp256k1<Signing>>` function for that till I noticed the global context type is incompatible.